irohad/pending_txs_storage/impl/pending_txs_storage_impl.cpp

```diff
@@ -64,10 +69,35 @@ PendingTransactionStorageImpl::getPendingTransactions(
   while (account_batches.batches.end() != batch_iterator
          and remaining_space >= batch_iterator->get()->transactions().size()) {
     auto &txs = batch_iterator->get()->transactions();
-    response.transactions.insert(
-        response.transactions.end(), txs.begin(), txs.end());
-    remaining_space -= txs.size();
-    ++batch_iterator;
+    // can be easily extended by adding more conditions if needed
+    if (first_tx_time or last_tx_time){
+      auto time_predicate = [&](auto &tx) {
```

Comment:
In previous PR: https://github.com/hyperledger/iroha/pull/1092/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.hpp
You were using `<`, but here You are using `<=` - it should be consistent, look:
```
        "AND (:first_tx_time::text IS NULL OR :first_tx_time<ts)",
          "AND (:last_tx_time::text IS NULL OR :last_tx_time>ts )",
```